Add types for chrome.i18n

ssorallen opened the pull request:
Add unimplemented types for [`chrome.i18n`][0]. Types are all based off
Chrome's documentation.

Addresses #1

[0]: https://developer.chrome.com/extensions/i18n

ssorallen committed:
Add types for chrome.i18n

Add unimplemented types for [`chrome.i18n`][0]. Types are all based off
Chrome's documentation.

Addresses #1

[0]: https://developer.chrome.com/extensions/i18n

Shraymonks commented on interfaces/i18n.js:
```diff
@@ -0,0 +1,14 @@
+type chrome$LanguageCode = string;
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (Array<chrome$LanguageCode>) => void): void,
```
Name the callback parameter?
```javascript
getAcceptLanguages(callback: (languages: Array<chrome$LanguageCode>) => void): void,
```

Shraymonks commented on interfaces/i18n.js:
```diff
@@ -0,0 +1,14 @@
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (Array<chrome$LanguageCode>) => void): void,
+  getMessage(messageName: string, ...substitutions?: Array<*>): ?string,
```
`substitutions` is just an array parameter, not rest parameters https://developer.chrome.com/extensions/i18n#method-getMessage

ssorallen replied:
Nice catch, thanks.

I'm using the API, and I missed the note that it wasn't rest params. It actually works with a single string, but more than one requires an Array. The examples even mention the single string case:

https://developer.chrome.com/apps/i18n#examples-getMessage

It seems best to document the example and the actual API even though their documentation doesn't agree.

What do you think?

Shraymonks commented on interfaces/i18n.js:
```diff
@@ -0,0 +1,14 @@
+  getUILanguage(): string,
+  detectLanguage(text: string, callback: (result: {
+    isReliable: boolean,
+    languages: {
```
`languages` is an array of these objects
https://developer.chrome.com/extensions/i18n#method-detectLanguage

Shraymonks reviewed (changes requested):
Thanks for your work on this!

ssorallen committed:
Match `getMessage` signature; extract types

`DetectedLanguage` and `LanguageDetectionResult` are names that appear
in documentation. Extract them as types to match the docs.

ssorallen commented:
Thanks for the feedback. I addressed each point.

Shraymonks commented on interfaces/i18n.js:
```diff
@@ -0,0 +1,18 @@
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (languages: Array<chrome$LanguageCode>) => void): void,
+  getMessage: (messageName: string, substitutions?: string | Array<string>) => string,
```
`getMessage` can return `undefined`

ssorallen replied:
Oh yes thanks. I mistakenly changed that.

Shraymonks commented on interfaces/i18n.js:
```diff
@@ -0,0 +1,18 @@
+
+type chrome$i18n = {
+  getAcceptLanguages(callback: (languages: Array<chrome$LanguageCode>) => void): void,
+  getMessage: (messageName: string, substitutions?: string | Array<string>) => string,
```
Change this back to the method annotation style
```js
getMessage(...): type,
```

ssorallen replied:
Will do.

ssorallen committed:
Prefer method syntax; match nullable return value

`getMessage` can return null if the given message is not found.

ssorallen commented:
@Shraymonks All set.

Shraymonks commented:
@ssorallen Thanks!
